packages/htmlviewer-extension/src/index.tsx

```diff
@@ -131,23 +133,34 @@ function activateHTMLViewer(
     });
   }
 
-  app.docRegistry.addWidgetFactory(factory);
-  factory.widgetCreated.connect((sender, widget) => {
-    // Track the widget.
-    void tracker.add(widget);
-    // Notify the widget tracker if restore data needs to update.
-    widget.context.pathChanged.connect(() => {
-      void tracker.save(widget);
-    });
-    // Notify the application when the trust state changes so it
-    // can update any renderings of the trust command.
-    widget.trustedChanged.connect(() => {
-      app.commands.notifyCommandChanged(CommandIDs.trustHTML);
-    });
+  // Fetch settings if possible.
+  const fetchSettings = settingRegistry
+    ? settingRegistry.load(HTML_VIEWER_PLUGIN_ID)
+    : Promise.reject(
```

Comment:
Since `ISettingRegistry` is declared as `optional` at the plugin level definition, a common pattern in the JupyterLab code base (and in extensions) is to first check for `settingRegistry` before using it.

Some alternative lab-based applications might be reusing the plugin without providing a `ISettingRegistry`.

For example in the `documentsearch-extension`:

https://github.com/jupyterlab/jupyterlab/blob/3291708449f7fe008daf0ba1be63c65f3dbc2bb1/packages/documentsearch-extension/src/index.ts#L113-L130

Reply:
Indeed this common pattern makes sense.  
I pushed a commit to try to implement it in this PR.  
Is it ok ?